### DISCUSSION TITLE
Maximising Batt life (and coolness) in IMU trigger mode?

### DISCUSSION THREAD

**Discussion by piersg** [gopro/labs#1204] [Q&A] [answered]
Sorry for slight repetition, I read an answer to this somewhere here on Github (or maybe just the internet) about a year ago, but I can't find it anymore.

We have a use case where a bunch of GoPros sit out in the Summer sun on surfboards for about 8 hours, and occasionally go out in the water - it's only this moving bit that's of interest, so IMU triggers are perfect, but ideally we'd also like to shut as much other stuff off while they're not recording to preserve the battery and also let things cool down.

I believe I've seen an answer from @dnewman-gpsw saying that actually shutting the sensor down is not supported, but you could put the whole GP into standby and wake periodically to check for motion - danged if I can find it now.

So my questions are:

- Is that !N sleep? or a shutdown? (IOW: is it possible to shut down fully for a set time period, or is that a one-way thing?)
- Do we know what the approximate power "cost" of the startup is (IOW: how frequently could I sleep and still save power? If we missed the first 10 seconds of an event, that might not be ideal) Obvs. IO can measure this IRL but if anyone has experience they can share, great.
- Or have things changed and there is a supported way to shut down the camera now?
- Any other ways to achieve what I'm after?

**Comment by dnewman-gpsw** [comment, marked as answer]
- Any sleep over 9 seconds will shutdown the camera. Same as powering off.
- Boot is not make more demanding than running the camera without recording. If you disable auto off, a camera will remain on from a full battery for around 3.5 hours (very rough estimate.)
When for camera boot up enough to test the gyro is using ~2-3W, this is with a 10% LCD brightness.  When it is encoding video 5W or more (depending on the video mode.) To go through the full boot, test gyro, and shutdown again is ~15s.  If you tested once per minute, you would have a 4X (60/15) increase in standby time, so that camera can be in standby for 14 hours. In the worst case you would loose up 60 seconds of footage. If the surfboards/camera are sitting the beach, you wouldn't lose much, as within an average 30s on them picked up they would be recording until put on the beach again.

```*BOOT="!Lbt"!SAVEbt==A0=A+g<l9!R3=A/9<A1!55RQ!S=A+g=A/2<A0.2!E+!55RQ$PMSG="gyro $A"!R-36```

This script will run for ~18 hours.

**Reply by dnewman-gpsw** [reply]
Any feedback?

**Reply by piersg** [reply]
I thought it would be presumptuous to hit you up for even more support, but if you're following up:

I'm using mostly GoPro 9s so I got as far as: needs to start !M, I think it needs to be !Lbt, I don't think there should a double equal after 'save' (all these are me guessing) and then it just got too complicated for me to really diagnose anything. Then some other year end jobs got in the way of further GoPro tinkering..

With the !M and !Lbt changed, it did say (something like) 'boot CMD = bt' and if I shut down it started up saying command was bt, but no IMU recording happened.

They would be running whatever firmware was current in June, if that helps (nothing stopping me from upgrading, that's just what was already there).

Thanks for checking in

**Reply by dnewman-gpsw** [reply]
Sorry, this command language is for GP2 cameras only, HERO9 doesn't have a way to test the IMU like this.

**Reply by piersg** [reply]
👍 we have a 12 which someone has borrowed, I'll give this a go when it's back

**Reply by piersg** [reply]
.. And are we saying that on the older cameras there is no way to say "do IMU trigger but with a timeout should nothing happen"? (Because I certainly can't find it).

I might get the 12 back tomorrow, but if not it will be after Jan 20 due to some summer holidays that don't overlap

**Reply by dnewman-gpsw** [reply]
Correct, need a newer model.

**Reply by piersg** [reply]
So a little later than promised, the colleague has returned our GoPro 12!

and .. I put the code above into the QR generator and the camera says, on powerup: "features activated: BOOT"; then the labs firmware message; then "Boot command: !bt".

But doesn't seem to ever record no matter how vigourously I shake it, and it powers down after roughly a minute, never to come back up.

It is now on the latest firmware, but was initially on 2.10 (ish) and it did even less (just a message after you scan the QR but nothing after boot IIRC).

I will leave it alone overnight and see if the battery is tellingly drained in the morning.

The project to record on the beach in the heat hasn't gotten up the last 2 summers, so this is more of an intellectual exercise now, but I'd love to get it working, you never know when it could come in handy

**Reply by dnewman-gpsw** [reply]
!Lbt is the command to want.

**Reply by piersg** [reply]
That's certainly working better now, but it's a little inconsistent (I need to gather more data to answer this more helpfully, but here's what I have so far). E.g. yesterday it more or less worked as advertised at the start of the day, but at some point the gyro started reading 7.x (when thr GP was stationary on a table) and it thus recorded continuously. 

~~This morning, after accidentally corrupting and reformatting the card (and scanning the QR again) the unit never gets to the "gyro" readout at the top, it just boots, display the full screen messages, then has "Labs processing" at the top for maybe a second or two then shuts down again.~~

Ah, I get it, it has to be moving at the point of wakeup. I'll have to figure out what the actual commands do so I can tweak the timing, but I think we can call this one answered
